pkg/utils/filter_map.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright © 2020 Doppler <support@doppler.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+import (
+	"strings"
+)
+
+// FilterMap returns a new map[string]T containing origMap keys that match aginst the keysToSelect
+func FilterMap[T any](origMap map[string]T, keysToSelect []string) map[string]T {
+	filteredMap := map[string]T{}
+
+	for _, keyToSelect := range keysToSelect {
+		key := strings.Trim(keyToSelect, " ")
```

Comment:
I was thinking the same and was undecided on whether or not to trim.

The reason I trimmed in the first place was because I thought it was providing some convenience if the user mistakenly added a space like such (when using quotes):
`doppler run --only-secrets "LOGGING , DB" --no-exit-on-missing-only-secrets  -- <command>`

But since you've pointed it out, I'm going to remove the `strings.Trim` line and make it so the users need to really pay attention to what they are typing.